linfa-trees/src/decision_trees/algorithm.rs

```diff
@@ -627,19 +635,101 @@ mod tests {
         let model = DecisionTree::params(2).max_depth(Some(2)).fit(&dataset);
 
         assert_eq!(&model.features(), &[8]);
+        assert_eq!(
+            &model.feature_importance(),
+            &[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0]
+        );
     }
 
     #[test]
     /// Small perfectly separable dataset test
     ///
     /// This dataset of three elements is perfectly using the second feature.
     fn perfectly_separable_small() {
-        let data = array![[1., 2., 3.5], [1., 2., 3.5], [1., 3., 3.5]];
+        let data = array![[1., 2., 3.], [1., 2., 3.5], [1., 3., 4.]];
         let targets = array![0, 0, 1];
 
         let dataset = Dataset::new(data.clone(), targets);
         let model = DecisionTree::params(2).max_depth(Some(1)).fit(&dataset);
 
         assert_eq!(&model.predict(data.clone()), &[0, 0, 1]);
     }
+
+    #[test]
+    /// Small toy dataset from sklearn
```

Comment:
Was there sklearn code used to generate this? Even if it's in python would be nice to have documentation on how to reproduce it.

Reply:
they were taken from here https://github.com/scikit-learn/scikit-learn/blob/0fb307bf39bbdacd6ed713c00724f8f871d60370/sklearn/tree/tests/test_tree.py#L67

Reply:
Ah nice!